web-common/src/features/dashboards/state-managers/selectors/measures.ts

```diff
@@ -145,10 +147,11 @@ export const getFilteredMeasuresAndDimensions = ({
       });
       if (skipMeasure) return;
       measures.add(measureName);
-      measure.referencedMeasures?.filter((refMes) => measures.add(refMes));
+      if (!measure.window) nonWindowMeasures.add(measureName);
```

Comment:
I'm having trouble following this function. 
1. What is a "filtered" measure and dimension?
2. I'm nervous about this line. Adding measures to the list of user-selected measures feels like exactly what caused the original bug. Double-checking... are you sure this is necessary?

Reply:
Adding these measures break the totals query, since we dont really get totals data anyway (`no data` can be seen) it is better to filter them out.

Would be great to add E2E once the setup PR is merged.

Reply:
It seems one way this function is overloaded is that it's preparing two different query requests: `totals` and `timeseries`, which each have a different set of measures & dimensions.

Can we split this into two functions:
1. `getMeasuresAndDimensionsForTimeSeriesRequest()`
2. `getMeasuresForTotalsRequest()`

Reply:
Updated the method. I dont think the method should differentiate between timeseries vs totals request. Added a param to include/exclude window measures.